src/cdk-experimental/menu/menu-group.spec.ts

```diff
@@ -0,0 +1,218 @@
+        .queryAll(By.directive(CdkMenuGroup))[3]
+        .injector.get(CdkMenuGroup)
+        .change.subscribe(spy);
+      menuItems[4].disabled = true;
+
+      menuItems[4].trigger();
+
+      expect(spy).toHaveBeenCalledTimes(0);
+    });
+
+    it('should not emit from sibling groups', () => {
+      const spies: Array<jasmine.Spy> = [];
+
+      fixture.debugElement.queryAll(By.directive(CdkMenuGroup)).forEach((group, index) => {
+        const spy = jasmine.createSpy(`cdkMenuGroup ${index} change spy`);
+        spies.push(spy);
+        group.injector.get(CdkMenuGroup).change.subscribe(spy);
+      });
+
+      menuItems[0].trigger();
+
+      expect(spies[1]).toHaveBeenCalledTimes(1);
+      expect(spies[2]).toHaveBeenCalledTimes(0);
+      expect(spies[3]).toHaveBeenCalledTimes(0);
+    });
+
+    it('should not emit on menuitem click', () => {
+      const spies: Array<jasmine.Spy> = [];
+
+      fixture.debugElement.queryAll(By.directive(CdkMenuGroup)).forEach((group, index) => {
+        const spy = jasmine.createSpy(`cdkMenuGroup ${index} change spy`);
+        spies.push(spy);
+        group.injector.get(CdkMenuGroup).change.subscribe(spy);
+      });
+
+      menuItems[7].trigger();
+
+      spies.forEach((spy) => expect(spy).toHaveBeenCalledTimes(0));
+    });
+  });
+});
+
+@Component({
+  selector: 'menu-with-groups',
+  template: `
+    <ul cdkMenu>
+      <!-- Checkbox group -->
+      <li role="none">
+        <ul cdkMenuGroup>
+          <li #first role="none">
+            <button checked="true" role="menuitemcheckbox" cdkMenuItem>
+              one
+            </button>
+          </li>
+          <li role="none">
+            <button role="menuitemcheckbox" cdkMenuItem>
+              two
+            </button>
+          </li>
+        </ul>
+      </li>
+      <!-- Radio group -->
+      <li role="none">
+        <ul cdkMenuGroup>
+          <li role="none">
+            <button checked="true" role="menuitemradio" cdkMenuItem>
+              three
+            </button>
+          </li>
+          <li role="none">
+            <button role="menuitemradio" cdkMenuItem>
+              four
+            </button>
+          </li>
+        </ul>
+      </li>
+      <!-- Radio group -->
+      <li role="none">
+        <ul cdkMenuGroup>
+          <li role="none">
+            <button role="menuitemradio" cdkMenuItem>
+              five
+            </button>
+          </li>
+          <li role="none">
+            <button role="menuitemradio" cdkMenuItem>
+              six
+            </button>
+          </li>
+        </ul>
+      </li>
+      <li role="none">
+        <ul cdkMenuGroup>
+          <li role="none">
+            <button cdkMenuItem>
+              seven
+            </button>
+          </li>
+          <li role="none">
+            <button cdkMenuItem>
+              eight
+            </button>
+          </li>
+        </ul>
+      </li>
+    </ul>
+  `,
+})
+class MenuGroups {}
```

Comment:
I think the smaller, focused test components from the menu-bar test suite would be better. (That way, it's easier to see the purpose of each test component, without the need for explanatory HTML comments.)

Reply:
ya so the menu/menu-bar tests I made small however here I also wanted to test interaction with groups next to each other hence why it's larger than normal.

Reply:
Oh, you were testing radio group independence? That's a good thing to test.

I still might consider splitting the checkbox group into a separate group, though, unless you need it in a test with radio groups.

Reply:
Exactly. When I was refactoring the circular reference issue it ended up catching a potential bug - so it was useful already.

Reply:
I would name this component something like `MenuWithMultipleSelectionGroups`. Ideally you would only use this component for tests that deal specifically with interactions related to having multiple selection-toggle controls. For more basic functionality (like change events), we could create a smaller test component